service/src/main/java/org/broadinstitute/listener/relay/inspectors/HeaderLoggerInspector.java

```diff
@@ -20,27 +24,69 @@ public class HeaderLoggerInspector implements RequestInspector {
   @Override
   public boolean inspectWebSocketUpgradeRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
-
-    logger.debug("WebSocket Upgrade Request: {}", relayedHttpListenerRequest.getUri());
-
-    logHeaders(relayedHttpListenerRequest.getHeaders());
-
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "WEBSOCKET_UPGRADE_REQUEST");
     return true;
   }
 
   @Override
   public boolean inspectRelayedHttpRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "HTTP_REQUEST");
+    return true;
+  }
+
+  @VisibleForTesting
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      OffsetDateTime requestTimestamp,
+      String prefix) {
+    var headers = relayedHttpListenerRequest.getHeaders();
+    var referer = headers.getOrDefault("Referer", "");
+    var origin = headers.getOrDefault("Origin", "");
+    var ua = headers.getOrDefault("User-Agent", "");
+    var sub = getTokenClaim(relayedHttpListenerRequest.getHeaders(), "sub").orElse("");
+
+    // get the idtyp so we can log if the request was made by a uami (i.e,. the value is "app")
+    var idTyp = getTokenClaim(relayedHttpListenerRequest.getHeaders(), "idtyp").orElse("");
```

Comment:
This claim probably works.. I think in apache we use the `xms_mirid` claim to determine if it's a UAMI (the claim contains the UAMI objectId).